pycodestyle.py

```diff
@@ -433,7 +474,8 @@ def extraneous_whitespace(logical_line):
         if text == char + ' ':
             # assert char in '([{'
             yield found + 1, "E201 whitespace after '%s'" % char
-        elif line[found - 1] != ',':
+        elif (line[found - 1] != ',' and
+              not (char == ':' and is_a_slice(line, found))):
```

Comment:
```suggestion
        elif (line[found - 1] != ',' 
              and not (char == ':' and is_a_slice(line, found))):
```
I was under the impression that 'pythonic' style puts the binary operator *after* a line break, not before.

--- Personal preference / opinion only below this line. Disregard at your pleasure ---

I personally also prefer to format multi-line conditions like this:

```python
        elif (
            line[found - 1] != ','
            and not (char == ':' and is_a_slice(line, found))
        ):
```

because it puts co-ordinate conditions in a visually co-ordinate structure, and also has dedicated lines to mark the bounds of the compound condition. (i.e. the `elif (` and `):` lines, visually positioned to encapsulate the conditions and indicate that they're subordinate to the `elif` construct itself.  It does take up a lot more vertical space though, so I can understand why some might dislike that.

Reply:
Guess this is the right pep for this:
https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

Reply:
@hoylemd I agree and prefer this coding style as well.
Yet W503 in TravisCI prevents me of doing so :

> pycodestyle.py:489:13: W503 line break before binary operator
> pycodestyle.py:512:13: W503 line break before binary operator

If @s-weigand suggestion to ignore W503 was accepted then we could proceed :)

Reply:
> I was under the impression that 'pythonic' style puts the binary operator _after_ a line break, not before.

This was change only a few years back (roughly when W504 was introduced and both were added to the default list). pycodestyle *itself* does not use this style of line breaks before binary operators and you should always adhere to the project's style.